### PR TITLE
docs: remove references to ADC environment variable

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14914)

### CODE REVIEW

File: google/cloud/spanner/benchmarks/README.md [hunk 1]
```diff
@@ -48,16 +48,9 @@ of this README, we assume the reader is familiar with the topic, and refer them
...
-`GOOGLE_APPLICATION_CREDENTIALS` environment variable to the path of this file.
-
-If you are running the benchmarks in a virtual machine, the library can
-automatically use the GCE instance service account (when you do **not** set
-`GOOGLE_APPLICATION_CREDENTIALS`). You may need to grant this service account
-permissions to work with Cloud Spanner. Examine the
-[spanner roles][spanner-roles-link] to chose a role for this account, the
-principal used to run these benchmark should have (at least) the permissions
-granted by the `roles/spanner.databaseAdmin` role.
+Examine the [spanner roles][spanner-roles-link] to chose a role for the
```
Comment (coryan) [inline google/cloud/spanner/benchmarks/README.md:51]: This one merits more attention.

File: google/cloud/universe_domain/demo/README.md [hunk 2]
```diff
@@ -32,29 +32,13 @@ These examples uses several different Google Cloud Platform (GCP) services. All
...
-library searches for a credentials file in the same location as the
-[Cloud SDK](https://cloud.google.com/sdk/). For more information about
-*Application Default Credentials*, see
-https://cloud.google.com/docs/authentication/production
-
-If the file found at `GOOGLE_APPLICATION_CREDENTIALS` contains the
-`universe_domain` field, that value is used to determine service endpoints. If
-the `universe_domain` field is not present, the default value of
-`googleapis.com` is used.
+[Authentication Overview][authentication-quickstart].
```
Comment (coryan) [inline google/cloud/universe_domain/demo/README.md:35]: Here too.

**Comment by coryan** [comment]
PTAL

**Review by piaxc — COMMENTED** [review]
Can you also change the link target for authentication-quickstart to https://cloud.google.com/docs/authentication/

Thanks for doing this!

**Comment by coryan** [comment]
> Can you also change the link target for authentication-quickstart to https://cloud.google.com/docs/authentication/

Done.

> Thanks for doing this!

No worries.

**Review by dbolduc — APPROVED** [review]
Thanks!

**Pull request merged**